sound/soc/intel/boards/sof_nau8825.c

```diff
@@ -581,24 +579,41 @@ static int sof_audio_probe(struct platform_device *pdev)
 	/* compute number of dai links */
 	sof_audio_card_nau8825.num_links = 1 + dmic_be_num + hdmi_num;
 
-	if (sof_nau8825_quirk & SOF_SPEAKER_AMP_PRESENT)
+	if (ctx->amp_type != CODEC_NONE)
 		sof_audio_card_nau8825.num_links++;
 
-	if (sof_nau8825_quirk & SOF_MAX98373_SPEAKER_AMP_PRESENT)
-		max_98373_set_codec_conf(&sof_audio_card_nau8825);
-	else if (sof_nau8825_quirk & SOF_RT1015P_SPEAKER_AMP_PRESENT)
-		sof_rt1015p_codec_conf(&sof_audio_card_nau8825);
-
 	if (sof_nau8825_quirk & SOF_SSP_BT_OFFLOAD_PRESENT)
 		sof_audio_card_nau8825.num_links++;
 
-	dai_links = sof_card_dai_links_create(&pdev->dev, ssp_codec, ssp_amp,
-					      dmic_be_num, hdmi_num);
+	dai_links = sof_card_dai_links_create(&pdev->dev, ctx->amp_type,
+					      ssp_codec, ssp_amp, dmic_be_num,
+					      hdmi_num);
 	if (!dai_links)
 		return -ENOMEM;
 
 	sof_audio_card_nau8825.dai_link = dai_links;
 
+	/* update codec_conf */
+	if (ctx->amp_type != CODEC_NONE) {
```

Comment:
why not remove this if and use it like
````c
switch (ctx->amp_type) {
                ...
		case CODEC_NONE:
                break;
                default:
                ...
}
````

Reply:
fixed